src/tools/rados/rados.cc

```diff
@@ -111,7 +111,8 @@ void usage(ostream& out)
 "                                    clean up a previous benchmark operation\n"
 "                                    default run-name is 'benchmark_last_metadata'\n"
 "   load-gen [options]               generate load on the cluster\n"
-"   listomapkeys <obj-name>          list the keys in the object map\n"
+"   listomapkeys <obj-name> [--start-key start-key]\n"
```

Comment:
@dongbula what's the use case of this option?  IIUC, `listomapkeys` walks through all the omap keys until there is no more of them. so how do you plan to use this option?

Reply:
hi, this option can be used in filtering from lot of omap keys. Such as, picking up one user's usage records upon a large usage shard in the object storage.RGW usage records prefix by user id or unix timestamp, this option will be helpful to analysis those records.

Reply:
hi, this option can be used in filtering from lot of omap keys. Such as, picking up one user's usage records upon a large usage shard in the object storage.RGW usage records prefix by user id or unix timestamp, this option will be helpful to analysis those records.

Reply:
@dongbula so what you are after is not a "start-key", but a prefix or a filter, am i correct?

Reply:
yes, but there is not native interface for prefix, so a start-key may also be a flexible choice

Reply:
no, we do have APIs for filtering omap by prefix
```c++
    int omap_get_vals2(const std::string& oid,
               const std::string& start_after,
               const std::string& filter_prefix,
               uint64_t max_return,
               std::map<std::string, bufferlist> *out_vals,
               bool *pmore);
```

Reply:
but it's interface for get vals, not for get keys

Reply:
`out_vals` contains both keys and values. am i missing anything?

Reply:
you meas the `listomapkeys ` can change rados call interface from omap_get_keys to omap_get_vals? it may case more data transmission in network.
```
} else if (strcmp(nargs[0], "listomapkeys") == 0) {
    if (!pool_name || nargs.size() < 2) {
      usage(cerr);
      return 1;
    }

    string last_read = start_key;
    bool more = true;
    do {
      set<string> out_keys;
      ret = io_ctx.**omap_get_keys2**(nargs[1], last_read, MAX_OMAP_BYTES_PER_REQUEST, &out_keys, &more);
      if (ret < 0) {
        cerr << "error getting omap key set " << pool_name << "/"
             << nargs[1] << ": "  << cpp_strerror(ret) << std::endl;
        return 1;
      }
```